addons/account_peppol/models/account_move.py

```diff
@@ -21,6 +21,19 @@ class AccountMove(models.Model):
         copy=False,
         readonly=True,
     )
+    ref = fields.Char(compute='_compute_payment_reference', readonly=False, store=True)
+
+    def _compute_payment_reference(self):
+        # EXTENDS account
+        super()._compute_payment_reference()
+        for move in self.filtered(lambda m: (
+            m.state != 'cancelled'
+            and m.move_type in ('out_invoice', 'out_refund')
+            and not m.ref
+        )):
+            company = move.company_id or move.journal_id.company_id
+            if company.is_account_peppol_participant:
```

Comment:
In my opinion, we should make sure that the `account_peppol_proxy_state` is active, because this way if a user just ticks the peppol checkbox in the settings without going through the registration, the ref will be generated for  them anyway, even thought they're not actually a peppol user